diff --git a/meson.build b/meson.build
index 50df973..d84357a 100644
--- a/meson.build
+++ b/meson.build
@@ -113,7 +113,7 @@ elif crypto == 'gcrypt'
 	crypto_dep += dependency('gpg-error', required : true)
 	conf.set('USE_GCRYPT', 1, description : 'If set use gcrypt crypto library.')
 elif crypto == 'openssl'
-	crypto_dep += dependency('openssl')
+	crypto_dep += dependency('openssl', static: true)
 	conf.set('USE_OPENSSL', 1, description : 'if set use openssl crypto library.')
 elif crypto == 'kernel'
 	crypto_dep += dependency('disabler-appears-to-disable-executable-build', required : false)
