diff --git a/Makefile b/Makefile
index b46f8ca..7e6c7e0 100644
--- a/Makefile
+++ b/Makefile
@@ -30,9 +30,12 @@ ALL_OBJS = $(addprefix obj/, $(filter-out $(REPLACED_OBJS), $(sort $(BASE_OBJS)
 LIBC_OBJS = $(filter obj/src/%,$(ALL_OBJS))
 LDSO_OBJS = $(filter obj/ldso/%,$(ALL_OBJS:%.o=%.lo))
 CRT_OBJS = $(filter obj/crt/%,$(ALL_OBJS))
+SKARNET_OBJS = $(wildcard obj/skalibs/*.lo)
+NSSS_OBJS = $(wildcard obj/nsss/*.lo)
+UTMPS_OBJS = $(wildcard obj/utmps/*.lo)
 
-AOBJS = $(LIBC_OBJS)
-LOBJS = $(LIBC_OBJS:.o=.lo)
+AOBJS = $(sort $(LIBC_OBJS) $(SKARNET_OBJS $(NSSS_OBJS)) $(UTMPS_OBJS))
+LOBJS = $(sort $(LIBC_OBJS:.o=.lo) $(SKARNET_OBJS) $(NSSS_OBJS) $(UTMPS_OBJS))
 GENH = obj/include/bits/alltypes.h obj/include/bits/syscall.h
 GENH_INT = obj/src/internal/version.h
 IMPH = $(addprefix $(srcdir)/, src/internal/stdio_impl.h src/internal/pthread_impl.h src/internal/locale_impl.h src/internal/libc.h)
