diff --git a/Makefile b/Makefile
index e8713982..7dfc9694 100644
--- a/Makefile
+++ b/Makefile
@@ -23,20 +23,6 @@ SUBDIR += libpe
 SUBDIR += isa  # ('isa' does not build on all platforms yet).
 .endif
 
-# Build tools after the libraries.
-SUBDIR += addr2line
-SUBDIR += ar
-SUBDIR += brandelf
-SUBDIR += cxxfilt
-SUBDIR += elfcopy
-SUBDIR += elfdump
-SUBDIR += findtextrel
-SUBDIR += ld
-SUBDIR += nm
-SUBDIR += readelf
-SUBDIR += size
-SUBDIR += strings
-
 # Build the test suites.
 .if exists(${.CURDIR}/test) && defined(WITH_TESTS) && ${WITH_TESTS} == "yes"
 SUBDIR += test
