diff --git a/source/includes/vst3sdk/pluginterfaces/base/funknown.cpp b/source/includes/vst3sdk/pluginterfaces/base/funknown.cpp
index 206a64a..916e63a 100644
--- a/source/includes/vst3sdk/pluginterfaces/base/funknown.cpp
+++ b/source/includes/vst3sdk/pluginterfaces/base/funknown.cpp
@@ -35,10 +35,6 @@
 
 #endif
 
-#if SMTG_OS_LINUX
-#include <ext/atomicity.h>
-#endif
-
 namespace Steinberg {
 
 //------------------------------------------------------------------------
@@ -72,7 +68,7 @@ int32 PLUGIN_API atomicAdd (int32& var, int32 d)
 #elif SMTG_OS_MACOS
 	return OSAtomicAdd32Barrier (d, (int32_t*)&var);
 #elif SMTG_OS_LINUX
-	__gnu_cxx::__atomic_add (&var, d);
+	__atomic_add_fetch(&var, d, __ATOMIC_ACQ_REL);
 	return var;
 #else
 #warning implement me!
