diff --git a/src/libunwind/build.rs b/src/libunwind/build.rs
index a24808b3250..9f3f5149b8d 100644
--- a/src/libunwind/build.rs
+++ b/src/libunwind/build.rs
@@ -106,9 +106,6 @@ mod llvm_libunwind {
         }
 
         if target_env == "musl" {
-            // use the same C compiler command to compile C++ code so we do not need to setup the
-            // C++ compiler env variables on the builders
-            cfg.cpp(false);
             // linking for musl is handled in lib.rs
             cfg.cargo_metadata(false);
             println!("cargo:rustc-link-search=native={}", env::var("OUT_DIR").unwrap());
diff --git a/src/libunwind/lib.rs b/src/libunwind/lib.rs
index 18d41be7739..fd97a764aa0 100644
--- a/src/libunwind/lib.rs
+++ b/src/libunwind/lib.rs
@@ -20,7 +20,7 @@ cfg_if::cfg_if! {
 
 #[cfg(target_env = "musl")]
 #[link(name = "unwind", kind = "static", cfg(target_feature = "crt-static"))]
-#[link(name = "gcc_s", cfg(not(target_feature = "crt-static")))]
+#[link(name = "unwind", cfg(not(target_feature = "crt-static")))]
 extern "C" {}
 
 #[cfg(target_os = "redox")]
