diff --git a/media/mtransport/third_party/nICEr/src/stun/stun.h b/media/mtransport/third_party/nICEr/src/stun/stun.h
index a32751d795..4ca9df6c33 100644
--- a/media/mtransport/third_party/nICEr/src/stun/stun.h
+++ b/media/mtransport/third_party/nICEr/src/stun/stun.h
@@ -47,6 +47,7 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <net/if_dl.h>
 #include <net/if_types.h>
 #else
+#include <net/if.h>
 #include <linux/if.h>
 #endif
 #ifndef BSD
